### PR TITLE
add missing chef files

### PR DESCRIPTION
**Pull request by MaximilianMeister**

### CODE REVIEW

**Comment by MaximilianMeister** [comment]
rebased

File: crowbar_framework/lib/crowbar/backup/restore.rb [hunk 1]
```diff
@@ -70,27 +79,39 @@ def restore_chef
...
+          "#{src_path}/."
+        else
+          src_path.to_s
+        end
+
+        system(
+          "sudo", "-i",
+          "cp", "-a",
+          src_string,
+          @data.join("crowbar", source).to_s,
```
Comment (rhafer) [inline crowbar_framework/lib/crowbar/backup/restore.rb]: You need to remove this line. The source is already in ```src_string```

File: crowbar_framework/lib/crowbar/backup/restore.rb [hunk 2]
```diff
@@ -40,6 +47,8 @@ def restore_chef
         begin
           [:nodes, :roles, :clients, :databags].each do |type|
             Dir.glob(@data.join("knife", type.to_s, "**", "*")).each do |file|
+              # skip client "crowbar"
+              next if type == :clients && file.basename.to_s =~ /^crowbar.json$/
```
Comment (rhafer) [inline crowbar_framework/lib/crowbar/backup/restore.rb]: ```file``` is not a Pathname here yet.

Just moving the next line in front of this one should help

File: crowbar_framework/lib/crowbar/backup/restore.rb (hunk 1 above)
Comment (MaximilianMeister) [inline crowbar_framework/lib/crowbar/backup/restore.rb]: sorry, manual rebase mistake

**Comment by rsalevsky** [comment]
Maybe it's better when we transform the file list into a hash and add this just as `key: value`.

**Comment by MaximilianMeister** [comment]
> Maybe it's better when we transform the file list into a hash and add this just as key: value.

I agree a Hash is always nicer, but I think it is fine as an Array as long as you don't have more than two values in each entry.

**Comment by MaximilianMeister** [comment]
rebased

File: crowbar_framework/lib/crowbar/backup/restore.rb [hunk 3]
```diff
@@ -70,27 +80,38 @@ def restore_chef
         true
       end
 
+      def restore_files(source, destination)
+        # keep the permissions of the files that are already in place
+        src_path = @data.join("crowbar", source)
+        dest_is_dir = system("sudo", "-i", "test", "-d", destination)
```
Comment (aplanas) [inline crowbar_framework/lib/crowbar/backup/restore.rb]: Ahm not sure, but `sudo -i` will put you in /root (in a different working directory that you are now)

File: crowbar_framework/lib/crowbar/backup/restore.rb [hunk 4]
```diff
@@ -70,27 +80,38 @@ def restore_chef
...
+        # If source and destination are both directories we just need to
+        # copy the contents of source, not the directory itself.
+        src_string = if dest_is_dir && src_path.directory?
+          "#{src_path}/."
+        else
+          src_path.to_s
+        end
+
+        system(
+          "sudo", "-i",
```
Comment (aplanas) [inline crowbar_framework/lib/crowbar/backup/restore.rb]: same here?

File: crowbar_framework/lib/crowbar/backup/restore.rb (hunk 3 above)
Comment (rhafer) [inline crowbar_framework/lib/crowbar/backup/restore.rb]: IIRC ```destination``` is an absolute path here. But I guess you're still right the ```-i``` shouldn't be needed here. (And in the other place)

**Comment by MaximilianMeister** [comment]
@aplanas @rhafer  dropped "-i"

**Comment by rhafer** [comment]
+1

File: crowbar_framework/lib/crowbar/backup/restore.rb [hunk 5]
```diff
@@ -70,27 +80,38 @@ def restore_chef
...
+          "#{src_path}/."
+        else
+          src_path.to_s
+        end
+
+        system(
+          "sudo",
+          "cp", "-a",
+          src_string,
+          destination
```
Comment (aplanas) [inline crowbar_framework/lib/crowbar/backup/restore.rb]: bit weird line layout?

**Comment by aplanas** [comment]
+1 for me

**Comment by rsalevsky** [comment]
+1

**Pull request merged**